tests/unit/etlng/ext/MPTTests.cpp

```diff
@@ -0,0 +1,164 @@
+constinit auto const kHASH3 = "6005B465CBBF7FA8E41AC0C0CD38491026D9411FCB7BA46E2AEBB3AF7654261D";
+
+auto
+createTestData()
+{
+    auto transactions = std::vector{
+        util::createTransaction(ripple::TxType::ttMPTOKEN_ISSUANCE_CREATE),  // not AUTHORIZE so will not be written
+        util::createTransaction(ripple::TxType::ttMPTOKEN_AUTHORIZE, kHASH, kTXN_META, kTXN_HEX),
+        util::createTransaction(ripple::TxType::ttAMM_CREATE),               // not MPT - will be filtered
+        util::createTransaction(ripple::TxType::ttMPTOKEN_ISSUANCE_CREATE),  // not unique - will be filtered
+    };
+
+    auto const header = createLedgerHeader(kLEDGER_HASH, kSEQ);
+    return etlng::model::LedgerData{
+        .transactions = std::move(transactions),
+        .objects = {},
+        .successors = {},
+        .edgeKeys = {},
+        .header = header,
+        .rawHeader = {},
+        .seq = kSEQ
+    };
+}
+
+auto
+createMultipleHoldersTestData()
+{
+    auto transactions = std::vector{
+        util::createTransaction(ripple::TxType::ttMPTOKEN_AUTHORIZE, kHASH, kTXN_META, kTXN_HEX),
+        util::createTransaction(ripple::TxType::ttMPTOKEN_AUTHORIZE, kHASH2, kTXN_META, kTXN_HEX),
+        util::createTransaction(ripple::TxType::ttMPTOKEN_AUTHORIZE, kHASH3, kTXN_META, kTXN_HEX)
+    };
+
+    auto const header = createLedgerHeader(kLEDGER_HASH, kSEQ);
+    return etlng::model::LedgerData{
+        .transactions = std::move(transactions),
+        .objects = {},
+        .successors = {},
+        .edgeKeys = {},
+        .header = header,
+        .rawHeader = {},
+        .seq = kSEQ
+    };
+}
+
+}  // namespace
+
+struct MPTExtTests : util::prometheus::WithPrometheus, MockBackendTest {
+protected:
+    MPTExt ext_{backend_};
+};
+
+TEST_F(MPTExtTests, OnLedgerDataFiltersAndWritesMPTs)
+{
+    auto const data = createTestData();
+
+    EXPECT_CALL(*backend_, writeMPTHolders).WillOnce([](auto const& holders) {
+        EXPECT_EQ(holders.size(), 1);  // Only AUTHORIZE is written in the end
+    });
+
+    ext_.onLedgerData(data);
+}
+
+TEST_F(MPTExtTests, OnInitialDataFiltersAndWritesMPTs)
+{
+    auto const data = createTestData();
+
+    EXPECT_CALL(*backend_, writeMPTHolders).WillOnce([](auto const& holders) {
+        EXPECT_EQ(holders.size(), 1);  // Only AUTHORIZE is written in the end
+    });
+
+    ext_.onInitialData(data);
+}
+
+TEST_F(MPTExtTests, OnInitialObjectWritesMPT)
+{
+    auto const data = util::createObjectWithMPT();
+
+    EXPECT_CALL(*backend_, writeMPTHolders).WillOnce([](auto const& holders) { EXPECT_EQ(holders.size(), 1); });
+
+    ext_.onInitialObject(kSEQ, data);
+}
+
+TEST_F(MPTExtTests, OnInitialDataWithMultipleHolders)
+{
+    auto const data = createMultipleHoldersTestData();
+
+    EXPECT_CALL(*backend_, writeMPTHolders).WillOnce([](auto const& holders) {
+        EXPECT_EQ(holders.size(), 3);  // Expect all three AUTHORIZE transactions
```

Comment:
It would be good to assert the actual holder account ID extracted like `EXPECT_EQ(holders[0], "ABCD...")`

Since all three mock transactions have the same metadata (if i'm understanding that correctly), then all three should have the same value as well